plugin/kubernetes/object/endpoint.go

```diff
@@ -44,12 +44,7 @@ type EndpointPort struct {
 func EndpointsKey(name, namespace string) string { return name + "." + namespace }
 
 // ToEndpoints converts an api.Endpoints to a *Endpoints.
-func ToEndpoints(obj interface{}) (*api.Endpoints, *Endpoints) {
-	end, ok := obj.(*api.Endpoints)
-	if !ok {
-		return nil, nil
-	}
-
+func ToEndpoints(end *api.Endpoints) (*api.Endpoints, *Endpoints) {
```

Comment:
I still don't get why we pass in `end` and then just return it.

Reply:
I mean, why do we need to return it?

Reply:
> I mean, why do we need to return it?

You're right.  I should do more refactoring here...